### PR TITLE
[python3.10] Build py310 images

### PR DESCRIPTION
**Pull request by scv119**
## Why are these changes needed?

Build python 3.10 images so we can run release tests.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by amogkam — APPROVED** [review]
Thanks, lgtm if tests pass! 

Also seems like some release test changes were accidentally pushed?

**Comment by amogkam** [comment]
@scv119 before merging this in, we also need to update the `ray/docker/retag-lambda/python_versions.txt` file.

**Review by mwtian — APPROVED** [review]

**Review by krfricke — APPROVED** [review]
Looks good, failing wheel build will be resolved once merged

**Pull request merged**
